pkg/subctl/cmd/diagnose/globalnet.go

```diff
@@ -0,0 +1,162 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package diagnose
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/submariner-io/submariner-operator/internal/cli"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd"
+	"github.com/submariner-io/submariner/pkg/globalnet/constants"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	mcsClientset "sigs.k8s.io/mcs-api/pkg/client/clientset/versioned"
+)
+
+func init() {
+	diagnoseCmd.AddCommand(&cobra.Command{
+		Use:   "globalnet",
+		Short: "Check globalnet configuration",
+		Long:  "This command checks globalnet configuration",
+		Run: func(command *cobra.Command, args []string) {
+			cmd.ExecuteMultiCluster(restConfigProducer, checkGlobalNet)
+		},
+	})
+}
+
+func checkGlobalNet(cluster *cmd.Cluster) bool {
+	status := cli.NewStatus()
+
+	if cluster.Submariner == nil {
+		status.Start(cmd.SubmMissingMessage)
+		status.EndWith(cli.Warning)
+
+		return true
+	}
+
+	status.Start("Checking Globalnet configuration")
+
+	if cluster.Submariner.Spec.GlobalCIDR != "" {
+		checkClusterGlobalEgressIps(cluster, status)
+		checkGlobalEgressIps(cluster, status)
+		checkGlobalIngressIps(cluster, status)
+
+		if status.HasFailureMessages() {
+			status.EndWith(cli.Failure)
+			return false
+		}
+
+		status.EndWithSuccess("Globalnet is enabled and properly configured")
+	} else {
+		status.EndWithSuccess("Globalnet is disabled")
+	}
```

Comment:
I think we should check for enabled above. Also we can simplify the code by using the new `reporter.Interface` methods.

```suggestion
	if cluster.Submariner == nil {
		status.Warning(cmd.SubmMissingMessage)
		return true
	}
	
	if cluster.Submariner.Spec.GlobalCIDR == "" {
	    status.Success("Globalnet is not installed - skipping")
	    return true
	}

	status.Start("Checking Globalnet configuration")
	defer status.End()

	return checkClusterGlobalEgressIps(cluster, status) &&
		   checkGlobalEgressIps(cluster, status) &&
		   checkGlobalIngressIps(cluster, status)
```